### PR TITLE
refactor: UI revindication - update UI hook to reuse on diamond

### LINKED ISSUE #731: Add UI diamond hook
After implementing the diamond proxy in the [contracts package](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts) most of our contracts (i.e. [facets](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/src/dollar/facets)) have the same address (which basically equals to the deployed `Diamond` contract). The only exception (regarding the address) are the [core contracts](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/src/dollar/core) which should be deployed with the own upgrade mechanism because function names from the [core contracts](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/src/dollar/core) overlap with the ones in the [facets](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/src/dollar/facets) and Diamond proxy doesn't support function clashing.

So basically this code:
```typescript
const managerAddress = "0x4e037B9A8Ce977462DA4E10Fc164363C827abfc6";
const managerContract = await getUbiquityManagerContract(managerAddress, provider);
```
is the same as this one:
```typescript
const diamondAddress = '0xbe0efAbc83686a81903C1D4a2515f8111e53B5Cb';
const managerContract = await getUbiquityManagerContract(diamondAddress, provider);
```

There is the [react hook](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx) that returns all of the contract instances

We should take all of the [facets](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/src/dollar/facets) and make sure that this [hook](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx) returns all of the contract instances (i.e. facets) by a single diamond address

We can either refactor the [existing hook](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx) or create a new one

This "return contracts" hook will be referenced in the next github issues (regarding the `dapp` package) where we will make sure, one by one, that all of the pages work fine with the diamond proxy

### PR DESCRIPTION
**Pull request by molecula451**
Resolves #731 

Alrighty! The hook is usable now.

### Some things

I have voided call reverts by  Adding a conditional to check for network ID:

### Fixes Call Reverts when loading
call

### Network ID
https://github.com/ubiquity/ubiquity-dollar/blob/5bf990c5624ab19c9cf194ff336bb17c2fe81850/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx#L35

This won't clog the UI with calls to the contract when the chain ID is unmatched, pretty clear huh?

```
yarn start:anvil
yarn deploy:development
yarn start (UI)
```

### Commit with anvil by default
yarn start:anvil
yarn deploy:development

use the commit
https://a7632410.ubiquity-dollar-5vd.pages.dev/

### CODE REVIEW

File: packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx [hunk 1]
```diff
@@ -29,9 +29,10 @@ export const ManagedContractsContextProvider: React.FC<ChildrenShim> = ({ childr
   const { provider } = useWeb3();
   const deployedContracts = useDeployedContracts();
   const [managedContracts, setManagedContracts] = useState<ManagedContracts>(null);
+  const NETWORK_ANVIL_ID = 31337;
 
   useEffect(() => {
-    if (deployedContracts && provider) {
+    if (deployedContracts && provider && provider.network.chainId === NETWORK_ANVIL_ID) {
```
Comment (molecula451) [inline packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx]: added to ignore call reverts

**Comment by molecula451** [comment]
@rndquu it'll be helpful to me that you add some line comments (on the files) (if possible) of some thoughts as the will help with further UI updates in regard to contract connectivity etc

said, because, i guess sushi pair isn't a thing right now

https://github.com/ubiquity/ubiquity-dollar/blob/be3784f8698702b9ecb2496a6ba0a9d099c883c4/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx#L82

File: packages/dapp/pages/index.tsx [hunk 2]
```diff
@@ -17,6 +17,9 @@ const index: FC = (): JSX.Element => {
   useEffectAsync(async () => {
     if (managedContracts != null) {
       try {
+        //DEBUG
+        console.log(managedContracts?.dollarToken.address, "MANAGED DOLLAR");
```
Comment (molecula451) [inline packages/dapp/pages/index.tsx:21]: hook in work

File: packages/dapp/pages/index.tsx [hunk 3]
```diff
@@ -17,6 +17,9 @@ const index: FC = (): JSX.Element => {
...
+        console.log(managedContracts);
```
Comment (molecula451) [inline packages/dapp/pages/index.tsx:22]: & hook fetching ALL the addresses at return

https://github.com/ubiquity/ubiquity-dollar/blob/c28cae99bd525b45fbcc7da9539966ca79c8f2b5/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx#L82

File: packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx [hunk 4]
```diff
@@ -79,7 +80,7 @@ async function connectManagerContracts(manager: ManagerFacet, provider: NonNulla
   const creditNftCalculator = manager.address;
   const sushiSwapPoolContract = getSushiSwapPoolContract(sushiSwapPool, provider);
 
-  const governanceMarket = getUniswapV2PairContract(await sushiSwapPoolContract.pair(), provider);
+  const governanceMarket = getUniswapV2PairContract(ZERO_ADDRESS, provider);
```
Comment (molecula451) [inline packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx:83]: @rndquu need some input on this one as this `governanceMarket` can be found in other parts of the UI

**Comment by molecula451** [comment]
@rndquu if you run yarn start:anvil && yarn deploy:development

https://173b50b8.ubiquity-dollar-5vd.pages.dev/

this commit will ask you to change the provider to hardhat automatically on  (MetaMask)

check the console and the hook should be fetching the addresses

File: packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx (hunk 1 above)
Comment (molecula451) [reply packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx]: call

**Comment by molecula451** [comment]
https://github.com/ubiquity/ubiquity-dollar/pull/763/commits/85e8bc4895fe25e4f0359c7f43cfd53e6b35036b updates `use-deployed-address.tsx` to enhace reading, in it's actual stage works but it looks unmanageable, this is a much better version for upcoming UI updates

**Comment by molecula451** [comment]
i think the `web3 provider` update may be in along with the #528

File: packages/dapp/components/lib/hooks/use-web-3.tsx [hunk 5]
```diff
@@ -39,7 +39,7 @@ const client = createClient(
     autoConnect: true,
     connectors: [
       new MetaMaskConnector({
-        chains: defaultChains,
+        chains: defaultChains.slice(),
```
Comment (molecula451) [inline packages/dapp/components/lib/hooks/use-web-3.tsx:42]: This does it! 

commit

You getting null on mainnet
You getting the contracts on hardhat (anvil)

Comment (molecula451) [reply packages/dapp/components/lib/hooks/use-web-3.tsx:42]: instad of showing unsupported network for anvil

**Comment by molecula451** [comment]
```
yarn start:anvil
yarn deploy:development
```

https://a7632410.ubiquity-dollar-5vd.pages.dev/

Connect as usual and switch to anvil connection

**Review by rndquu — CHANGES_REQUESTED** [review]
1. This [hook](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx) was a mess before the current PR. It still remains a mess. I expected it to be refactored similar to:
```js
// get diamond with all of the facets
const diamondAddress = getLatestDeployedDiamond();
const diamondContract = new ethers.Contract(diamondAddress, DIAMOND_ABI, provider);
// get contracts deployed apart from the diamond (i.e. not part of the diamon's facets)
const dollarContract = new ethers.Contract(dollarAddress, DOLLAR_ABI, provider);
const creditContract = new ethers.Contract(creditAddress, CREDIT_ABI, provider);

return {
  // diamond facets
  accessControl: new ethers.Contract(diamondAddress, ACCESS_CONTROL_ABI, provider),
  collectableDustContract: new ethers.Contract(diamondAddress, COLLECTABLE_DUST_ABI, provider),
  ...other facets
  // separately deployed contracts
  dollarContract,
  creditContract,
  ...other separately deployed contracts
};
```

This way bounty hunters would see the overall project structure (diamond + separate contracts) and it would be easier for them to fix UI.
2. This [hook](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx) does not return all of the [facets](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/src/dollar/facets)
3. Contract naming mismatch ([example](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx#L99): should be [stakingFormulas](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/dollar/facets/StakingFormulasFacet.sol))

File: packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx (hunk 4 above)
Comment (0x4007) [reply packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx:83]: `governanceMarket` is supposed to be the SushiSwap `uAD-UBQ` market but I think we should migrate to Uniswap. Why is this reset to `0x0`? The market currently exists.

**Comment by molecula451** [comment]
> Member




> 1. This [hook](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx) was a mess before the current PR. It still remains a mess. I expected it to be refactored similar to:
> 
> ```js
> // get diamond with all of the facets
> const diamondAddress = getLatestDeployedDiamond();
> const diamondContract = new ethers.Contract(diamondAddress, DIAMOND_ABI, provider);
> // get contracts deployed apart from the diamond (i.e. not part of the diamon's facets)
> const dollarContract = new ethers.Contract(dollarAddress, DOLLAR_ABI, provider);
> const creditContract = new ethers.Contract(creditAddress, CREDIT_ABI, provider);
> 
> return {
>   // diamond facets
>   accessControl: new ethers.Contract(diamondAddress, ACCESS_CONTROL_ABI, provider),
>   collectableDustContract: new ethers.Contract(diamondAddress, COLLECTABLE_DUST_ABI, provider),
>   ...other facets
>   // separately deployed contracts
>   dollarContract,
>   creditContract,
>   ...other separately deployed contracts
> };
> ```
> 
> This way bounty hunters would see the overall project structure (diamond + separate contracts) and it would be easier for them to fix UI. 2. This [hook](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx) does not return all of the [facets](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/src/dollar/facets) 3. Contract naming mismatch ([example](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx#L99): should be [stakingFormulas](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/dollar/facets/StakingFormulasFacet.sol))

If we do `new.ethers.Contract`

```js
return {
  // diamond facets
  accessControl: new ethers.Contract(diamondAddress, ACCESS_CONTROL_ABI, provider),
  collectableDustContract: new ethers.Contract(diamondAddress, COLLECTABLE_DUST_ABI, provider),
  ...other facets
  // separately deployed contracts
  dollarContract,
  creditContract,
  ...other separately deployed contracts
};
```

We would not be relying on this [contract component](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/utils/contracts.ts) that is in charge of take abi etc.. maybe something similar according to your suggestion?

File: packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx (hunk 4 above)
Comment (molecula451) [reply packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx:83]: this was until clarification, not really resseting to a zero address, more input is welcoming

**Comment by 0x4007** [comment]
> We would not be relying on this [contract component](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/utils/contracts.ts) that is in charge of take abi etc.. maybe something similar according to your suggestion?

Type cast it? Perhaps I'm not understanding the issue.

```ts
// get diamond with all of the facets
const diamondAddress = getLatestDeployedDiamond();
const diamondContract = new ethers.Contract(diamondAddress, DIAMOND_ABI, provider) as DiamondContract;
// get contracts deployed apart from the diamond (i.e. not part of the diamon's facets)
const dollarContract = new ethers.Contract(dollarAddress, DOLLAR_ABI, provider) as DollarToken;
const creditContract = new ethers.Contract(creditAddress, CREDIT_ABI, provider) as CreditToken;

```

**Comment by molecula451** [comment]
> > We would not be relying on this [contract component](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/utils/contracts.ts) that is in charge of take abi etc.. maybe something similar according to your suggestion?
> 
> Type cast it? Perhaps I'm not understanding the issue.
> 
> ```ts
> // get diamond with all of the facets
> const diamondAddress = getLatestDeployedDiamond();
> const diamondContract = new ethers.Contract(diamondAddress, DIAMOND_ABI, provider) as DiamondContract;
> // get contracts deployed apart from the diamond (i.e. not part of the diamon's facets)
> const dollarContract = new ethers.Contract(dollarAddress, DOLLAR_ABI, provider) as DollarToken;
> const creditContract = new ethers.Contract(creditAddress, CREDIT_ABI, provider) as CreditToken;
> ```


I think it's cleaner to keep to the fetch by the shorthand function component, i'm wondering since rndquu suggestion shows `new.ethers.Contract` , maybe he means something similar, while preserving same artifacts, components

**Comment by molecula451** [comment]
> Type cast it? Perhaps I'm not understanding the issue.
> 

no worries @pavlovcik , your input is always welcome and necessary (when possible and even if you dont understand), we make things better that way

**Comment by rndquu** [comment]
> > Member
> 
> > 1. This [hook](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx) was a mess before the current PR. It still remains a mess. I expected it to be refactored similar to:
> > 
> > ```js
> > // get diamond with all of the facets
> > const diamondAddress = getLatestDeployedDiamond();
> > const diamondContract = new ethers.Contract(diamondAddress, DIAMOND_ABI, provider);
> > // get contracts deployed apart from the diamond (i.e. not part of the diamon's facets)
> > const dollarContract = new ethers.Contract(dollarAddress, DOLLAR_ABI, provider);
> > const creditContract = new ethers.Contract(creditAddress, CREDIT_ABI, provider);
> > 
> > return {
> >   // diamond facets
> >   accessControl: new ethers.Contract(diamondAddress, ACCESS_CONTROL_ABI, provider),
> >   collectableDustContract: new ethers.Contract(diamondAddress, COLLECTABLE_DUST_ABI, provider),
> >   ...other facets
> >   // separately deployed contracts
> >   dollarContract,
> >   creditContract,
> >   ...other separately deployed contracts
> > };
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This way bounty hunters would see the overall project structure (diamond + separate contracts) and it would be easier for them to fix UI. 2. This [hook](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx) does not return all of the [facets](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/src/dollar/facets) 3. Contract naming mismatch ([example](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/lib/hooks/contracts/use-manager-managed.tsx#L99): should be [stakingFormulas](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/dollar/facets/StakingFormulasFacet.sol))
> 
> If we do `new.ethers.Contract`
> 
> ```js
> return {
>   // diamond facets
>   accessControl: new ethers.Contract(diamondAddress, ACCESS_CONTROL_ABI, provider),
>   collectableDustContract: new ethers.Contract(diamondAddress, COLLECTABLE_DUST_ABI, provider),
>   ...other facets
>   // separately deployed contracts
>   dollarContract,
>   creditContract,
>   ...other separately deployed contracts
> };
> ```
> 
> We would not be relying on this [contract component](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/utils/contracts.ts) that is in charge of take abi etc.. maybe something similar according to your suggestion?

Yes, we could also refactor this [file](https://github.com/ubiquity/ubiquity-dollar/blob/21ffe14e37d9ee626ce8e2fcca8a9f3817bf8a72/packages/dapp/components/utils/contracts.ts) (or create a new one)

The main point is that there should be a single place in the code which returns all of our contracts:
1. Separate contract for each of the diamond's [facets](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/src/dollar/facets)
2. [Other](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/src/dollar/core) separately deployed contracts (Dollar, Credit, etc...)

File: packages/dapp/components/lib/hooks/use-web-3.tsx (hunk 5 above)
Comment (molecula451) [reply packages/dapp/components/lib/hooks/use-web-3.tsx:42]: this one, it's going to be refactored but not on this PR, on this PR i will leave as it, fetching for all the available networks

the refactor is to make sure it doesn't look for anvil on live production network and just when dev or debug variable is activated

**Comment by molecula451** [comment]
> The main point is that there should be a single place in the code which returns all of our contracts:

@rndquu This is the task of the [ManagerFacet](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/dollar/facets/ManagerFacet.sol) address! so we basically have to refactor **ManagerFacet** with the other contract facets to bring the rest to the front end

but one thing take a look at this bug!

If we want to bring the [bancorFormulaAddress](https://github.com/ubiquity/ubiquity-dollar/blob/7e39594538f29922977806752352ee86246ef895/packages/contracts/src/dollar/facets/ManagerFacet.sol#L407) (just an example) (with the **ManagerFacet** contract in it's actual stage) to the front end

we will get
front

[We will find out that the address it's not set
](https://github.com/ubiquity/ubiquity-dollar/blob/7e39594538f29922977806752352ee86246ef895/packages/contracts/src/dollar/Diamond.sol#L53C10-L53C10)

Even tho it was already there from older commits! so I pressume, this will happen with any other new addition to the **ManagerFacet** as pivot to other facets

Which lead to the task to design and **debug**, iterate the contracts, this will take a bit, i wouldn't like to stale this PR nor to keep it drafted while debugging on design, so feel free to update the task description a bit or to not close the issue but commit this PR as there are good changes here,  maybe recheck? while i work on the update for this contracts or as last let me know if we are drafting this a bit while i debug

List of facets that needs to be pivoted:


#### Some of the facets that needs to be pivoted

1. AccessControl (unsure why we want this on the front end maybe for debug?)
2. CollectableDustFacet
3.  CurveDollarIncentiveFacet
4. DollarMintExcessFacet
5. BondingCurveFacet

**Comment by rndquu** [comment]
> so we basically have to refactor ManagerFacet with the other contract facets to bring the rest to the front end

I don't understand. How exactly we should refactor the `ManagerFacet` contract and why?

> If we want to bring the [bancorFormulaAddress](https://github.com/ubiquity/ubiquity-dollar/blob/7e39594538f29922977806752352ee86246ef895/packages/contracts/src/dollar/facets/ManagerFacet.sol#L407) (just an example) (with the ManagerFacet contract in it's actual stage) to the front end

Why can't we get it this way?

```
const diamondAddress = "0x01";
const managerContract = new ethers.Contract(diamondAddress, MANAGER_FACET_ABI, provider);
```

> Some of the facets that needs to be pivoted

What do you mean by "pivoted"?

**Comment by molecula451** [comment]
> I don't understand. How exactly we should refactor the `ManagerFacet` contract and why?

take a look a the ManagerFacet how it brings the facets that are already available on the hook (and remember this manager itself it's a facet)

> Why can't we get it this way?
> 
> ```
> const diamondAddress = "0x01";
> const managerContract = new ethers.Contract(diamondAddress, MANAGER_FACET_ABI, provider);
> ```

We are doing this. Manager becomes the main diamond to fetch the addresses
also this hook is spread everywhere on the UI. Deprecating it, it means re-touch the whole UI and clean at all.

but the concern of calling you was not that. The debug on the contracts to add the new facets to the front end is

We are just re-using this hook (remember it was only null) and the diamond main address it's fetched [here](https://github.com/molecula451/ubiquity-dollar/blob/3b186b9a936e803c468a784bbe4182019e6cdc62/packages/dapp/components/lib/hooks/contracts/use-deployed-contracts.tsx#L10)

Which is after **yarn deploy:development** 

Diamond: **0xbe0efAbc83686a81903C1D4a2515f8111e53B5Cb**

> What do you mean by "pivoted"?

To turn in, to make use of. The types artifacts are not generated within it's supposely to be 
```ts
ManagerFacet.getBondingCurveFacet()
```

this type it's not available anywhere for e.g. 
This one either
```ts
ManagerFacet.CollectableDustFacetAddress()
```
Bring it alone, it's not what we want

**Again**. on the initial design it's ManagerFacet it's assumed to be the main diamond, that's why you get revert calls like the above, otherwise you would get method not available

You can't have the types from the diamond ABI, it's part of the manager facet, that consumes [libAppStorage addresses](https://github.com/molecula451/ubiquity-dollar/blob/development/packages/contracts/src/dollar/libraries/LibAppStorage.sol)

[e.g Facet address](https://github.com/molecula451/ubiquity-dollar/blob/5db0d8efe45199906a3569805b7b1dfd650b672f/packages/contracts/src/dollar/libraries/LibAppStorage.sol#L17)
[e.g Facet on ManagerFacet contract](https://github.com/molecula451/ubiquity-dollar/blob/5db0d8efe45199906a3569805b7b1dfd650b672f/packages/contracts/src/dollar/facets/ManagerFacet.sol#L375C24-L375C24)

Something similar with the other facets that are not available! so that types to a "pivot" main contract becomes available in this case as the initial design (ManagerFacet)

**Comment by rndquu** [comment]
@molecula451 

This [issue](https://github.com/ubiquity/ubiquity-dollar/issues/731) implies creating a **single file** which returns:
1. All of the contracts that exist as the diamond facets
2. All of the contracts that are not part of the diamond

We don't need to refactor the whole UI in the current issue, neither refactoring solidity contracts. Imagine that the repo is empty and we need the code that fetches all of the project's contracts in the UI.

**Comment by molecula451** [comment]
> @molecula451 
> 
> This [issue](https://github.com/ubiquity/ubiquity-dollar/issues/731) implies creating a **single file** which returns:
> 1. All of the contracts that exist as the diamond facets
> 2. All of the contracts that are not part of the diamond
> 
> We don't need to refactor the whole UI in the current issue, neither refactoring solidity contracts. Imagine that the repo is empty and we need the code that fetches all of the project's contracts in the UI.

Creating a file won't make use of this Hook at all. Please take a bit of to digest the last answer while i debug things on my end and lets draw further points

**Comment by rndquu** [comment]
> The main point is that there should be a single place in the code which returns all of our contracts

> @rndquu This is the task of the [ManagerFacet](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/dollar/facets/ManagerFacet.sol) address! so we basically have to refactor ManagerFacet with the other contract facets to bring the rest to the front end

No, this is not the task of the [ManagerFacet](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/dollar/facets/ManagerFacet.sol) to return all of the contracts. In the early days there was a contract `UbiquityManager` which was responsible for storing contract addresses. Now we moved to the diamond proxy and `UbiquityManager` became `ManagerFacer` where most of the methods are obsolete because all of them return the same diamond address. For example, this [method](https://github.com/ubiquity/ubiquity-dollar/blob/7e39594538f29922977806752352ee86246ef895/packages/contracts/src/dollar/facets/ManagerFacet.sol#L113), this [method](https://github.com/ubiquity/ubiquity-dollar/blob/7e39594538f29922977806752352ee86246ef895/packages/contracts/src/dollar/facets/ManagerFacet.sol#L79), this [method](https://github.com/ubiquity/ubiquity-dollar/blob/7e39594538f29922977806752352ee86246ef895/packages/contracts/src/dollar/facets/ManagerFacet.sol#L153), all of them are setting the same diamond address right now which doesn't make sense. They should be either refactored, either deleted (which is part of another task).

We shouldn't build the frontend code that returns all of the contracts around the `ManagerFacet` but rather around the `Diamond` and separately deployed contracts (Dollar, Credit, etc...).

**Comment by molecula451** [comment]
> > The main point is that there should be a single place in the code which returns all of our contracts
> 
> > @rndquu This is the task of the [ManagerFacet](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/dollar/facets/ManagerFacet.sol) address! so we basically have to refactor ManagerFacet with the other contract facets to bring the rest to the front end
> 
> No, this is not the task of the [ManagerFacet](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/dollar/facets/ManagerFacet.sol) to return all of the contracts. In the early days there was a contract `UbiquityManager` which was responsible for storing contract addresses. Now we moved to the diamond proxy and `UbiquityManager` became `ManagerFacer` where most of the methods are obsolete because all of them return the same diamond address. For example, this [method](https://github.com/ubiquity/ubiquity-dollar/blob/7e39594538f29922977806752352ee86246ef895/packages/contracts/src/dollar/facets/ManagerFacet.sol#L113), this [method](https://github.com/ubiquity/ubiquity-dollar/blob/7e39594538f29922977806752352ee86246ef895/packages/contracts/src/dollar/facets/ManagerFacet.sol#L79), this [method](https://github.com/ubiquity/ubiquity-dollar/blob/7e39594538f29922977806752352ee86246ef895/packages/contracts/src/dollar/facets/ManagerFacet.sol#L153), all of them are setting the same diamond address right now which doesn't make sense. They should be either refactored, either deleted (which is part of another task).
> 
> We shouldn't build the frontend code that returns all of the contracts around the `ManagerFacet` but rather around the `Diamond` and separately deployed contracts (Dollar, Credit, etc...).

Making sense but this should imply deprecating the ManagerFacet hook which is spread across many part of the UI, or we ignore this hook for now and bring the the diamond in a new different way, like i fetched addresses on the last PR to local-data, without this hook

**Comment by 0x4007** [comment]
@rndquu perhaps sketching/rendering visual diagrams could be helpful for complex specifications like what we are starting to see here in `ubiquity-dollar`

An iPad Pro with Apple Pencil has gotten me through many moments like these.

**Comment by molecula451** [comment]
Yes as a last, i think we are moving this forward in subsequently issues, the thing is the original issue was about the "hook" but if we're deprecating this hook, then we have to revamp many part of the UI to avoid clashing. But in either way, should strive to keep the UI actively on deploys, while activelely refactoring it

**Comment by rndquu** [comment]
> @rndquu perhaps sketching/rendering visual diagrams could be helpful for complex specifications like what we are starting to see here in `ubiquity-dollar`
> 
> An iPad Pro with Apple Pencil has gotten me through many moments like these.

I will create a new wiki page with the protocol architecture when we have all of the contracts [upgradable](https://github.com/ubiquity/ubiquity-dollar/discussions/730)

**Comment by rndquu** [comment]
Closing this one in favour of https://github.com/ubiquity/ubiquity-dollar/pull/771

@molecula451 pls feel free to open separate PRs for each bug that the current PR solves

**Comment by rndquu** [comment]
> > @rndquu perhaps sketching/rendering visual diagrams could be helpful for complex specifications like what we are starting to see here in `ubiquity-dollar`
> > An iPad Pro with Apple Pencil has gotten me through many moments like these.
> 
> I will create a new wiki page with the protocol architecture when we have all of the contracts [upgradable](https://github.com/ubiquity/ubiquity-dollar/discussions/730)

https://github.com/ubiquity/ubiquity-dollar/wiki/23.-Smart-Contracts-Architecture

**Pull request closed without merging**